service-tests/bundlephobia.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const Joi = require('joi')
+const ServiceTester = require('./runner/service-tester')
+
+const t = new ServiceTester({
+  id: 'bundlephobia', title: 'NPM package bundle size',
+})
+
+const formats = {
+  A: '/bundlephobia/:type/:package.:format',
+  B: '/bundlephobia/:type/:package/:version.:format',
+  C: '/bundlephobia/:type/@:scope/:package.:format',
+  D: '/bundlephobia/:type/@:scope/:package/:version.:format',
+}
+
+const withoutErrorsMin = {
+  A: '/min/preact.json',
+  B: '/min/preact/8.2.7.json',
+  C: '/min/@cycle/core.json',
+  D: '/min/@cycle/core/7.0.0.json',
+}
+
+const withoutErrorsGzip = {
+  A: '/gzip/preact.json',
+  B: '/gzip/preact/8.2.7.json',
+  C: '/gzip/@cycle/core.json',
+  D: '/gzip/@cycle/core/7.0.0.json',
+}
+
+const withoutErrorSizesMin = {
+  A: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  B: '8.94 kB',
+  C: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  D: '3.51 kB',
+}
+
+const withoutErrorSizesGzip = {
+  A: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  B: '3.58 kB',
+  C: Joi.string().regex(/^\d+[.]?\d+ kB$/),
+  D: '1.23 kB',
+}
+
+const noExistPackages = {
+  A: '/min/some-no-exist.json',
+  B: '/min/some-no-exist/1.0.0.json',
+  C: '/min/@some-no-exist/some-no-exist.json',
+}
+
+Object.keys(formats).forEach(format => {
+  const withoutErrorMin = withoutErrorsMin[format]
+  const withoutErrorSizeMin = withoutErrorSizesMin[format]
+
+  const withoutErrorGzip = withoutErrorsGzip[format]
+  const withoutErrorSizeGzip = withoutErrorSizesGzip[format]
```

Comment:
This is nice and DRY, though it's a bit difficult to follow what you're doing. Also not sure it's necessary to be exhaustive in testing these combinations.

Could I suggest a couple ways of making it easier to understand? One would be to reduce this to 5 success cases (four min and one gzip) and 1–3 failure cases, and lay out each test explicitly. The other would be to basically invert your data objects, making an array of objects with `format`, `withoutErrorMin`, etc. as keys.